apisix/plugins/server-info.lua

```diff
@@ -132,51 +125,107 @@ local function get_server_info()
 end
 
 
+local function set(key,value,ttl, modified_index)
+    local res_new, err = core.etcd.atomic_set(key, value, ttl, modified_index)
+    if not res_new then
+        core.log.error("failed to set server_info: ", err)
+        return 503, {error_msg = err}
+    end
+
+    lease_id = res_new.body.lease_id
+
+    -- set or update lease_id
+    local ok, err = internal_status:set("lease_id", lease_id)
+    if not ok then
+        core.log.error("failed to set lease_id to shdict: ", err)
+        return 503, {error_msg = err}
+    end
+
+    return 200, nil
+end
```

Comment:
It looks like the two steps should be in sync, and we can use lock in the next PR to keep them in sync.